src/settingsdialog.cpp

```diff
@@ -262,3 +266,73 @@ void SettingsDialog::addSourcePathPage()
         Settings::instance()->setSourceCodePaths(sourcePathPage->sourcePaths->items().join(colon));
     });
 }
+
+void SettingsDialog::addSSHPage()
+{
+    auto page = new QWidget(this);
+    auto item = addPage(page, tr("SSH Settings"));
+    item->setHeader(tr("SSH Settings Page"));
+    item->setIcon(QIcon::fromTheme(QStringLiteral("preferences-system-windows-behavior")));
+    sshSettingsPage->setupUi(page);
+    sshSettingsPage->messageWidget->hide();
+    sshSettingsPage->errorWidget->hide();
+
+    auto configGroup = KSharedConfig::openConfig()->group("SSH");
+    sshSettingsPage->deviceConfig->setChildWidget(
+        sshSettingsPage->deviceSettings,
+        {sshSettingsPage->username, sshSettingsPage->hostname, sshSettingsPage->options});
+    sshSettingsPage->deviceConfig->setConfigGroup(configGroup);
+
+    connect(sshSettingsPage->copySshKeyButton, &QPushButton::pressed, this, [this] {
+        auto* copyKey = new QProcess(this);
+
+        auto path = sshSettingsPage->sshCopyIdPath->text();
+        if (path.isEmpty()) {
+            path = QStandardPaths::findExecutable(QStringLiteral("ssh-copy-id"));
+        }
+        if (path.isEmpty()) {
+            sshSettingsPage->messageWidget->setText(tr("Could not find ssh-copy-id"));
+            sshSettingsPage->messageWidget->show();
+            return;
+        }
+
+        if (qEnvironmentVariableIsSet("SSH_ASKPASS")) {
+            auto env = QProcessEnvironment::systemEnvironment();
```

Comment:
why do we have to set this, isn't the system env used by default and that will inherit the ssh_askpass value then, no?

Reply:
right, this piece of code is useless

Reply:
Actually this brakes if the `SSH_ASKPASS` and `SSH_ASKPASS_REQUIRE` are not set. In this case the user is prompted on the command line, which may not be accessable.

Reply:
I don't have `_REQUIRE` set in my env and it works. If `SSH_ASKPASS` isn't set, we can set it to `ksshaskpass` (and error out if that isn't available)